internal/celoapi/backend.go

```diff
@@ -37,55 +37,60 @@ type CeloAPIBackend struct {
 	exchangeRatesCache *lru.Cache[common.Hash, common.ExchangeRates]
 }
 
-func (b *CeloAPIBackend) getContractCaller(ctx context.Context, atBlock common.Hash) (*contracts.CeloBackend, error) {
+func (b *CeloAPIBackend) getContractCaller(ctx context.Context, blockNumOrHash rpc.BlockNumberOrHash) (*contracts.CeloBackend, error) {
 	state, _, err := b.Backend.StateAndHeaderByNumberOrHash(
 		ctx,
-		rpc.BlockNumberOrHashWithHash(atBlock, false),
+		blockNumOrHash,
 	)
 	if err != nil {
-		return nil, fmt.Errorf("retrieve state for block hash %s: %w", atBlock.String(), err)
+		return nil, fmt.Errorf("retrieve state for block hash %s: %w", blockNumOrHash.String(), err)
 	}
 	return &contracts.CeloBackend{
 		ChainConfig: b.Backend.ChainConfig(),
 		State:       state,
 	}, nil
 }
 
-func (b *CeloAPIBackend) GetFeeBalance(ctx context.Context, atBlock common.Hash, account common.Address, feeCurrency *common.Address) (*big.Int, error) {
-	cb, err := b.getContractCaller(ctx, atBlock)
+func (b *CeloAPIBackend) GetFeeBalance(ctx context.Context, blockNumOrHash rpc.BlockNumberOrHash, account common.Address, feeCurrency *common.Address) (*big.Int, error) {
+	cb, err := b.getContractCaller(ctx, blockNumOrHash)
 	if err != nil {
 		return nil, err
 	}
 	return contracts.GetFeeBalance(cb, account, feeCurrency), nil
 }
 
-func (b *CeloAPIBackend) GetExchangeRates(ctx context.Context, atBlock common.Hash) (common.ExchangeRates, error) {
-	cachedRates, ok := b.exchangeRatesCache.Get(atBlock)
-	if ok {
-		return cachedRates, nil
+func (b *CeloAPIBackend) GetExchangeRates(ctx context.Context, blockNumOrHash rpc.BlockNumberOrHash) (common.ExchangeRates, error) {
+	blockHash, isHash := blockNumOrHash.Hash()
+	if isHash {
+		cachedRates, ok := b.exchangeRatesCache.Get(blockHash)
+		if ok {
+			return cachedRates, nil
+		}
 	}
-	cb, err := b.getContractCaller(ctx, atBlock)
+	cb, err := b.getContractCaller(ctx, blockNumOrHash)
 	if err != nil {
 		return nil, err
 	}
 	er, err := contracts.GetExchangeRates(cb)
 	if err != nil {
 		return nil, err
 	}
-	b.exchangeRatesCache.Add(atBlock, er)
+	if isHash {
+		b.exchangeRatesCache.Add(blockHash, er)
```

Comment:
I wonder if we should add some tests for the caching here.

Reply:
I removed the whole LRU cache 514de826181d. The usefulness should be decided at a later point, and in general this
should probably be implemented on a lower level anyways.